### PR TITLE
Escape Double Quotes in tao_idl PP Args on Windows

### PR DESCRIPTION
**Pull request by iguessthislldo**
To fix issue with #1621 on the scoreboards.

### CODE REVIEW

**Comment by iguessthislldo** [comment]
... And vcpkg is kaput for a Windows specific PR. Perfect.

File: TAO/TAO_IDL/driver/drv_preproc.cpp [hunk 1]
```diff
@@ -147,14 +151,60 @@ DRV_cpp_putarg (const char *str)
...
     }
-  else
+  else if (str)
     {
-      DRV_arglist[DRV_argcount++] =
-        ACE::strnew (ACE_TEXT_CHAR_TO_TCHAR (str));
+#ifdef ACE_WIN32
+      // Escape Doublequotes on Windows
+      size_t quote_count = 0;
+      for (const char* quote = ACE_OS::strchr (str, '"'); quote; quote = ACE_OS::strchr (quote, '"'))
```
Comment (mitza-oci) [inline TAO/TAO_IDL/driver/drv_preproc.cpp:167]:
```suggestion
      for (const char *quote = ACE_OS::strchr (str, '"'); quote; quote = ACE_OS::strchr (quote, '"'))
```
also below on 179

File: TAO/TAO_IDL/driver/drv_preproc.cpp [hunk 2]
```diff
@@ -147,14 +151,60 @@ DRV_cpp_putarg (const char *str)
...
+            {
+              ACE_TCHAR *to = buf;
+              for (const char* from = str; *from; ++from)
+                {
+                  if (*from == '"')
+                    {
+                      *to = ACE_TEXT ('\\');
+                      ++to;
+                    }
+                  *to = *from;
```
Comment (mitza-oci) [inline TAO/TAO_IDL/driver/drv_preproc.cpp:186]: These are (potentially) different types, char and tchar.  Does this assignment always work?  At least add a comment?

Comment (iguessthislldo) [reply TAO/TAO_IDL/driver/drv_preproc.cpp:186]: Assuming `char` is ASCII and `wchar` encoding is ASCII-based it should just should widen the value like an int cast, but yeah that's probably assuming too much. Probably needs to convert the whole str first.

Comment (iguessthislldo) [reply TAO/TAO_IDL/driver/drv_preproc.cpp:186]: Actually I realized that's insane to work with `ACE_TCHAR` when I didn't have to and did the opposite of converting it first by converting it at the very end.

**Review by mitza-oci — APPROVED** [review]

File: TAO/TAO_IDL/driver/drv_preproc.cpp [hunk 3]
```diff
@@ -138,22 +140,68 @@ DRV_cpp_putarg (const char *str)
...
     }
-  else
+
+  DRV_arglist[DRV_argcount++] = ACE::strnew (ACE_TEXT_CHAR_TO_TCHAR (replace ? replace : str));
+
+  if (replace)
     {
-      DRV_arglist[DRV_argcount++] =
-        ACE::strnew (ACE_TEXT_CHAR_TO_TCHAR (str));
+      delete replace;
```
Comment (jwillemsen) [inline TAO/TAO_IDL/driver/drv_preproc.cpp:203]: New coverity error, shouldn't this be delete[] @iguessthislldo

Comment (iguessthislldo) [reply TAO/TAO_IDL/driver/drv_preproc.cpp:203]: Yes: #1633 and #1634.

**Pull request merged**